src/Build/BackEnd/Components/RequestBuilder/ITaskBuilder.cs

```diff
@@ -52,5 +53,11 @@ internal interface ITaskBuilder
         /// <param name="cancellationToken">The cancellation token used to cancel processing of the task.</param>
         /// <returns>A Task representing the work to be done.</returns>
         Task<WorkUnitResult> ExecuteTask(TargetLoggingContext targetLoggingContext, BuildRequestEntry requestEntry, ITargetBuilderCallback targetBuilderCallback, ProjectTargetInstanceChild task, TaskExecutionMode mode, Lookup lookupForInference, Lookup lookupForExecution, CancellationToken cancellationToken);
+
+        /// <summary>
+        /// Sets the task environment on the TaskExecutionHost for use with IMultiThreadableTask instances.
+        /// </summary>
+        /// <param name="taskEnvironment">The task environment to set, or null to clear.</param>
+        void SetTaskEnvironment(TaskEnvironment taskEnvironment);
```

Comment:
Please check if this method is needed and you can't do the same in ExecuteTask ->BuildRequestEntry